packages/react-component-library/src/components/Timeline/context/reducer.ts

```diff
@@ -89,20 +92,60 @@ export function buildCalendar({
   }
 }
 
+function transformOptions(
+  options: TimelineOptions,
+  currentScaleOption: TimelineScaleOption
+) {
+  return {
+    ...options,
+    endDate: options.endDate ? currentScaleOption.to : null,
+    startDate: currentScaleOption.from,
+  }
+}
+
 export function reducer(
   state: TimelineState,
   action: TimelineAction
 ): TimelineState | never {
-  const { scale } = action
-
-  return {
-    ...state,
-    ...buildCalendar(scale),
-    currentScaleOption: scale,
-    options: {
-      ...state.options,
-      startDate: scale.from,
-      endDate: state.options.endDate ? scale.to : null,
-    },
+  switch (action.type) {
+    case TIMELINE_ACTIONS.CHANGE_WIDTH:
+      /* eslint-disable no-case-declarations */
+      const scaleOptions = initialiseScaleOptions(state.options, action.width)
+      const currentScaleIndex =
+        state.currentScaleIndex ||
+        findIndex(scaleOptions, ({ isDefault }) => isDefault)
```

Comment:
Is there a reason to use the lodash helper over the native implementation?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex